src/gb_merkle_trees.erl

```diff
@@ -141,6 +142,30 @@ merkle_proof_node(Key, {InnerKey, _, Left, Right}) ->
             {node_hash(Left), merkle_proof_node(Key, Right)}
     end.
 
+-spec verify_merkle_proof(key(), value(), Root::hash(), merkle_proof()) ->
+                                 {ok, verified} | {error, Reason} when
```

Comment:
Why not simply `ok` instead of `{ok, verified}`?

Reply:
Strictly speaking, `ok` is ok as well.

Having to return the reason of the error, I attempted to adhere to the spec that - I found - callers usually expect i.e. `{ok, Result::term()} | {error, Reason::term()}`.

Reply:
In this case `ok` can however act as a result itself. The pattern `ok | {error, Reason :: term()}` can be found for example in [Mnesia](http://erlang.org/doc/man/mnesia.html).

Reply:
Addressed by new commit. CI is running.

Thanks for providing the Mnesia example.